File: ios/Classes/Channel/Error/WalletStripeError.swift
//
//  WalletStripeError.swift
//
//
//  Created by Miguel Ángel Soto González on 20/10/25.
//

import Flutter

enum WalletStripeError {
    case standard(details: String)
    case missingArgument(method: String)

    var rawValue: FlutterError {
        switch self {
        case .standard(let details):
            return FlutterError(
                code: "UNKNOWN",
                message: "Unknown Error",
                details: details
            )
        case .missingArgument(let method):
            return FlutterError(
                code: "MISSING_ARGUMENTS",
                message: "Missing Argument",
                details: "A required argument is missing from the method \(method)"
            )
        }
    }
}

File: android/src/main/kotlin/com/rudo/hybrid_stripe_payments/channel/error/WalletStripeError.kt
package com.rudo.hybrid_stripe_payments.channel.error

sealed class WalletStripeError {
    data class Standard(val details: String?) : WalletStripeError()
    data class MissingArgument(val method: String) : WalletStripeError()

    fun code(): String {
        return when (this) {
            is Standard -> "UNKNOWN"
            is MissingArgument -> "MISSING_ARGUMENTS"
        }
    }

    fun message(): String {
        return when (this) {
            is Standard -> "Unknown Error"
            is MissingArgument -> "Missing Argument"
        }
    }

    fun details(): String {
        return when (this) {
            is Standard -> details ?: "-"
            is MissingArgument -> "A required argument is missing from the method $method"
        }
    }
}
